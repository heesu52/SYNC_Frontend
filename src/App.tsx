import {
  BrowserRouter as Router,
  Routes,
  Route,
  Navigate,
} from 'react-router-dom';
import Login from './router/Login';
import SignUp from './router/SignUp';
import ModifyUser from './router/user/Modify';
import Home from './router/Home';
import NotFound from './router/NotFound';
import FindId from './router/user/FindId';
import FindPassword from './router/user/FindPassword';

const isLogin = true;

function App() {
  return (
    <Routes>
      <Route path="/" element={isLogin ? <Home /> : <Navigate to="login" />} />
      <Route path="/login" element={<Login />}></Route>
      <Route path="/signup" element={<SignUp />}></Route>
      <Route path="/:userId/modify" element={<ModifyUser />}></Route>
      <Route path="/user/find/userId" element={<FindId />}></Route>
      <Route path="/user/find/password" element={<FindPassword />}></Route>
      <Route path={'*'} element={<NotFound />}></Route>
    </Routes>
  );
}

export default App;
