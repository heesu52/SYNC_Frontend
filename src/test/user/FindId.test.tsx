import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import App from '../../App';

test('FindId 컴포넌트에 이메일 입력란이 포함되어있다.', () => {
  // Act
  render(
    <MemoryRouter initialEntries={['/user/find/userId']}>
      <App />
    </MemoryRouter>
  );

  // Assert
  const emailInput = screen.getByPlaceholderText('이메일');
  expect(emailInput).toBeInTheDocument();
});

test.each([
  {
    linkText: '아이디를 알고 있나요? 비밀번호 찾으러 가기',
    role: 'find-password',
  },
  { linkText: '회원가입 하러 가기', role: 'signup' },
  { linkText: '로그인으로 돌아가기', role: 'login' },
])(
  'FindId 컴포넌트에 $linkText Link를 클릭하면 role이 $role인 페이지로 경로가 변경된다.',
  async ({ linkText, role }) => {
    // Act
    render(
      <MemoryRouter initialEntries={['/user/find/userId']}>
        <App />
      </MemoryRouter>
    );

    //Assert
    const linkElement = screen.getByText(linkText, { exact: false });
    expect(linkElement).toBeInTheDocument();

    fireEvent.click(linkElement);
    const container = await screen.findByRole(role);
    expect(container).toBeVisible();
  }
);
