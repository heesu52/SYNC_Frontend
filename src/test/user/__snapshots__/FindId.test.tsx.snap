// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FindId /> 1`] = `
<div>
  <main>
    <section>
      <h1>
        Hinc
      </h1>
    </section>
    <section>
      <p>
        아이디가 기억나지 않나요?
      </p>
      <p>
        이메일 주소를 입력하시면 이메일로 계정에 다시 엑세스 할 수 잇는 아이디를 보내드립니다.
      </p>
      <form>
        <input
          placeholder="이메일"
          type="text"
        />
        <input
          type="submit"
          value="로그인 링크 보내기"
        />
      </form>
      <a
        href="/user/find/password"
      >
        아이디를 알고 있나요? 비밀번호 찾으러 가기 &gt;
      </a>
      <p>
        또는
      </p>
      <div>
        <a
          href="/signup"
        >
          회원가입 하러 가기
        </a>
      </div>
    </section>
    <article>
      <a
        href="/login"
      >
        로그인으로 돌아가기
      </a>
    </article>
  </main>
</div>
`;
